tests/unit-tests/jobs/models/jobExperimentsHandler-test.js

```diff
@@ -49,70 +53,87 @@ describe('Job experiments handler tests', function () {
-        clock.tick(3000);
-        experimentsManagerGetStub.callCount.should.eql(1);
-        experimentsManagerInsertStub.callCount.should.eql(2);
-        experimentsManagerInsertStub.args[0][1].should.eql(jobId);
-        experimentsManagerInsertStub.args[0][2].should.eql(experiment.id);
-        (experimentsManagerInsertStub.args[0][4] - experimentsManagerInsertStub.args[0][3]).should.eql(60000);
-        clock.restore();
-    });
+        it('set chaos experiments with same experiment in different times', async () => {
+            const experiment = generateExperiment();
+            experimentsManagerInsertStub.resolves();
+            experimentsManagerGetStub.resolves([experiment]);
+            const jobExperiments = [
+                {
+                    experiment_id: experiment.id,
+                    start_after: 1000
+                },
+                {
+                    experiment_id: experiment.id,
+                    start_after: 2000
+                }
+            ];
+            const jobId = uuid();
+            const clock = sinon.useFakeTimers();
+            clock.tick(1000);
+            await jobExperimentHandler.setChaosExperimentsIfExist(jobId, jobExperiments);
+            clock.tick(3000);
+            experimentsManagerGetStub.callCount.should.eql(1);
+            experimentsManagerInsertStub.callCount.should.eql(2);
+            experimentsManagerInsertStub.args[0][1].should.eql(jobId);
+            experimentsManagerInsertStub.args[0][2].should.eql(experiment.id);
+            (experimentsManagerInsertStub.args[0][4] - experimentsManagerInsertStub.args[0][3]).should.eql(60000);
+            clock.restore();
+        });
 
-    it('set chaos experiments with no experiments', async () => {
-        await jobExperimentHandler.setChaosExperimentsIfExist(uuid(), []);
-        experimentsManagerGetStub.callCount.should.eql(1);
-        experimentsManagerInsertStub.callCount.should.eql(0);
+        it('set chaos experiments with no experiments', async () => {
+            await jobExperimentHandler.setChaosExperimentsIfExist(uuid(), []);
+            experimentsManagerGetStub.callCount.should.eql(1);
+            experimentsManagerInsertStub.callCount.should.eql(0);
+        });
+    });
+    describe('stop chaos experiments for job', function () {
+        const jobExperimentHandlerRewire = rewire('../../../../src/jobs/models/kubernetes/jobExperimentsHandler');
+        it('set chaos experiments with 2 experiments', async () => {
+            const jobId = uuid();
+            const map = new Map();
+            const timeouts = [setTimeout(() => true, 10000), setTimeout(() => true, 50000)];
+            map.set(jobId, timeouts);
+            jobExperimentHandlerRewire.__set__('jobIdsToTimeouts', map);
+            stopJobExperimentsByJobIdStub.resolves();
+            await jobExperimentHandlerRewire.stopChaosExperimentsForJob(jobId);
```

Comment:
any chance we can check the experiments were added to the map before we do the stop job